handlers/gate/main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+)
+
+const (
+	baseBranch = "master"
+)
+
+// Config ...
+type Config struct {
+	BypassGate bool `env:"BYPASS_GATE"`
+}
+
+// BuildParamsModel ...
+type BuildParamsModel struct {
+	Branch                   string                     `json:"branch"`
+	Tag                      string                     `json:"tag"`
+	CommitHash               string                     `json:"commit_hash"`
+	CommitMessage            string                     `json:"commit_message"`
+	WorkflowID               string                     `json:"workflow_id"`
+	BranchDest               string                     `json:"branch_dest"`
+	PullRequestID            string                     `json:"pull_request_id"`
+	PullRequestRepositoryURL string                     `json:"pull_request_repository_url"`
+	PullRequestMergeBranch   string                     `json:"pull_request_merge_branch"`
+	PullRequestHeadBranch    string                     `json:"pull_request_head_branch"`
+	Environments             []EnvironmentVariableModel `json:"environments"`
+	BranchDestRepoOwner      string                     `json:"branch_dest_repo_owner"`
+	BranchRepoOwner          string                     `json:"branch_repo_owner"`
+}
+
+// EnvironmentVariableModel ...
+type EnvironmentVariableModel struct {
+	MappedTo string `json:"mapped_to"`
+	Value    string `json:"value"`
+	IsExpand bool   `json:"is_expand"`
+}
+
+func failf(format string, v ...interface{}) {
+	log.Errorf(format, v...)
+	os.Exit(1)
+}
+
+func rebuildAPICall() (string, error) {
+	buildParams := BuildParamsModel{
+		Branch:                   os.Getenv("BITRISE_GIT_BRANCH"),
+		Tag:                      os.Getenv("BITRISE_GIT_TAG"),
+		CommitHash:               os.Getenv("BITRISE_GIT_COMMIT"),
+		CommitMessage:            "Rebuilding with manually accepted step-info change.",
+		WorkflowID:               "pr_check",
+		BranchDest:               os.Getenv("BITRISEIO_GIT_BRANCH_DEST"),
+		PullRequestID:            os.Getenv("PULL_REQUEST_ID"),
+		PullRequestRepositoryURL: os.Getenv("BITRISEIO_PULL_REQUEST_REPOSITORY_URL"),
+		PullRequestMergeBranch:   os.Getenv("BITRISEIO_PULL_REQUEST_MERGE_BRANCH"),
+		PullRequestHeadBranch:    os.Getenv("BITRISEIO_PULL_REQUEST_HEAD_BRANCH"),
+		Environments: []EnvironmentVariableModel{
+			EnvironmentVariableModel{
+				MappedTo: "BYPASS_GATE",
+				Value:    "true",
+			},
+		},
+	}
+
+	buildParamsBytes, err := json.Marshal(buildParams)
+	if err != nil {
+		return "", fmt.Errorf("rebuildAPICall: failed to marshal build params, buildParams: %+v, %v", buildParams, err)
+	}
+
+	return fmt.Sprintf(`$ curl https://app.bitrise.io/app/a0bac497f75e1490/build/start.json --data '{"hook_info":{"type":"bitrise","build_trigger_token":" < insert_build_trigger_token > "},
```

Comment:
We might consider factoring the app slug to an env var instead of hard coding it in a public repo.

In case of a future bitrise.io vulnerability, it might be used for malicious purposes.